pkg/apis/core/v1alpha1/helper/shootstate_list.go

```diff
@@ -94,3 +96,36 @@ func (g *GardenerResourceDataList) Upsert(data *gardencorev1alpha1.GardenerResou
 	}
 	*g = append(*g, *data)
 }
+
+// ResourceDataList is a list of ResourceData
+type ResourceDataList []gardencorev1alpha1.ResourceData
+
+// Delete deletes an item from the list
+func (r *ResourceDataList) Delete(ref *autoscalingv1.CrossVersionObjectReference) {
```

Comment:
Can you add some small unit tests for these new functions?

Reply:
Done